### PR TITLE
Add Nexus configuration

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
Add configuration for Nexus deploy automation
Utilize new Nexus Staging plugin to handle, what would be otherwise manual, the process of closing and releasing repo for non-snapshot artifacts.

related to https://github.com/eclipse/che/issues/16043

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -292,6 +292,18 @@
         <pluginManagement>
             <!-- All plugins ordered by shortname (antrun, assembly ...) -->
             <plugins>
+            	<plugin>
```
Comment (sparkoo) [inline pom.xml]: formatting looks a bit off

File: pom.xml [hunk 2]
```diff
@@ -292,6 +292,18 @@
...
+				    <groupId>org.sonatype.plugins</groupId>
+				    <artifactId>nexus-staging-maven-plugin</artifactId>
+				    <version>1.6.7</version>
```
Comment (skabashnyuk) [inline pom.xml:298]: please put the plugin version into the variable.

**Comment by skabashnyuk** [comment]
BTW why do we need this change?

**Comment by mkuznyetsov** [comment]
@skabashnyuk this PR is part of the effort to make automated deply to nexus during release process (issue https://github.com/eclipse/che/issues/16043)
I realized I forgot to add necessary cico* script changes originally, so here they are

**Pull request merged**